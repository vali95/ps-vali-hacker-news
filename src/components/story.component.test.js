import React from 'react'
import {shallow} from 'enzyme'

import Story from './story.component'

describe('Story Component',()=>{
    
    it("Snapshot Test",()=>{ 
        const wrapper = shallow(<Story/>)       
        expect(wrapper).toMatchSnapshot()
    })
    it('upvote button exists',()=>{
        const wrapper = shallow(<Story/>) 
        expect(wrapper.find('.upvote-button')).toHaveLength(1)
    })
    it('upvote button click',()=>{
        const buttonClickMock = jest.fn()
        
        const wrapper = shallow(<Story castVote={buttonClickMock} />)

        wrapper.find('.upvote-button').simulate('click')
        expect(buttonClickMock.mock.calls.length).toBe(1)
        
        
    })
    
    it('Hide Link exists',()=>{
        const wrapper = shallow(<Story/>)
        expect(wrapper.find('.hide-button')).toHaveLength(1)
    })
    it('hide button click',()=>{
        const hideButtonClickMock  = jest.fn()
        const wrapper = shallow(<Story  hide={hideButtonClickMock}/>)
        wrapper.find('.hide-button').simulate('click')
        expect(hideButtonClickMock.mock.calls.length).toBe(1)
        
    })
        
    
})

describe('Testing Timestamp conversion',()=>{
    
    it('seconds',()=>{
        const created_at_i = (Date.now())/1000;
        const wrapper = shallow(<Story created_at_i={created_at_i}/>)        
        expect(wrapper.find('.dim').at(2).text()).toMatch(/Second/)
    })
    it('Minutes',()=>{
        const created_at_i = (Date.now()/1000)-(5*60);
        const wrapper = shallow(<Story created_at_i={created_at_i}/>)        
        expect(wrapper.find('.dim').at(2).text()).toMatch(/Minute/)
    })
    it('Hours',()=>{
        const created_at_i = (Date.now()/1000)-(5*60*60);
        const wrapper = shallow(<Story created_at_i={created_at_i}/>)        
        expect(wrapper.find('.dim').at(2).text()).toMatch(/Hour/)
    })
    it('Days',()=>{
        const created_at_i = (Date.now()/1000)-(5*24*60*60);
        const wrapper = shallow(<Story created_at_i={created_at_i}/>)        
        expect(wrapper.find('.dim').at(2).text()).toMatch(/Day/)
    })
    it('Weeks',()=>{
        const created_at_i = (Date.now()/1000)-(2*7*24*60*60);
        const wrapper = shallow(<Story created_at_i={created_at_i}/>)        
        expect(wrapper.find('.dim').at(2).text()).toMatch(/Week/)
    })
    it('Months',()=>{
        const created_at_i = (Date.now()/1000)-(5*30*24*60*60);
        const wrapper = shallow(<Story created_at_i={created_at_i}/>)        
        expect(wrapper.find('.dim').at(2).text()).toMatch(/Month/)
    })
    it('Years',()=>{
        const created_at_i = (Date.now()/1000)-(5*365*24*60*60);
        const wrapper = shallow(<Story created_at_i={created_at_i}/>)        
        expect(wrapper.find('.dim').at(2).text()).toMatch(/Year/)
    })

})