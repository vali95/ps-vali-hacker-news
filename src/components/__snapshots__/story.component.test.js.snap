// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Story Component Snapshot Test 1`] = `
<tr>
  <td />
  <td />
  <td>
    <button
      className="upvote-button"
      onClick={[Function]}
      type="button"
    >
      <i
        className="fas fa-caret-up"
      />
    </button>
  </td>
  <td>
    <span
      className="news"
    >
      <span
        className="title"
      />
       
      <a
        className="dim"
      >
        (
        )
      </a>
      <span
        className="dim"
      >
        by
      </span>
       
      <span
        className="author"
      />
      <span
        className="dim"
      >
        NaN Year(s) ago
      </span>
      <button
        className="hide-button"
        onClick={[Function]}
      >
        [ hide ]
      </button>
    </span>
  </td>
</tr>
`;
